hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/ReadBufferManager.java

```diff
@@ -116,6 +118,7 @@ void queueReadAhead(final AbfsInputStream stream, final long requestedOffset, fi
       buffer.setRequestedLength(requestedLength);
       buffer.setStatus(ReadBufferStatus.NOT_AVAILABLE);
       buffer.setLatch(new CountDownLatch(1));
+      buffer.setTracingContext(tracingContext);
 
       Integer bufferIndex = freeList.pop();  // will return a value, since we have checked size > 0 already
```

Comment:
In eviction code we had set tracing context to null ? its missing.

Reply:
added